expansion/src/main/java/no/unit/nva/expansion/model/cristin/CristinOrganization.java

```diff
@@ -1,21 +1,55 @@
 package no.unit.nva.expansion.model.cristin;
 
+import static java.util.Objects.nonNull;
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import java.net.URI;
 import java.util.List;
+import java.util.Map;
 import lombok.Getter;
 import no.unit.nva.commons.json.JsonSerializable;
-import no.unit.nva.model.Organization;
+import nva.commons.core.SingletonCollector;
 
 @Getter
 public class CristinOrganization implements JsonSerializable {
 
     public static final String PART_OF = "partOf";
+    public static final String ID = "id";
+    public static final String LABELS = "labels";
+    @JsonProperty(ID)
+    private final URI id;
     @JsonProperty(PART_OF)
-    public final List<Organization> partOf;
+    private final List<CristinOrganization> partOf;
+    @JsonProperty(LABELS)
+    private final Map<String, String> labels;
 
     @JsonCreator
-    public CristinOrganization(@JsonProperty(PART_OF) List<Organization> partOf) {
+    public CristinOrganization(@JsonProperty(ID) URI id,
+                               @JsonProperty(PART_OF) List<CristinOrganization> partOf,
+                               @JsonProperty(LABELS) Map<String, String> labels) {
+        this.id = id;
         this.partOf = partOf;
+        this.labels = labels;
+    }
+
+    @JsonIgnore
+    public CristinOrganization getTopLevelOrg() {
```

Comment:
Don't we get the topLevelOrganization from the cristin-proxy these days?

Reply:
We do, but in some cases when we have organization on "low" level (as example underenhet av fakultet), we get fakultet as topLevelOrg and this faculty again has its topLevelOrg which is main institution. Look at JSON above as example, and I am not sure how "deep" it goes, so by this while loop we can find topLevelOrg. 

```
{
  "@context" : "https://bibsysdev.github.io/src/organization-context.json",
  "type" : "Organization",
  "id" : "https://api.dev.nva.aws.unit.no/cristin/organization/184.13.14.0",
  "labels" : {
    "nb" : "Institutt for biomedisin"
  },
  "partOf" : [ {
    "type" : "Organization",
    "id" : "https://api.dev.nva.aws.unit.no/cristin/organization/184.13.0.0",
    "labels" : {
      "nb" : "Det medisinske fakultet"
    },
    "partOf" : [ {
      "type" : "Organization",
      "id" : "https://api.dev.nva.aws.unit.no/cristin/organization/184.0.0.0",
      "labels" : {
        "en" : "University of Bergen",
        "nb" : "Universitetet i Bergen"
      }
    } ]
  } ]
}
```